Library/Homebrew/utils/curl.rb

```diff
@@ -19,6 +20,8 @@ def curl(*args)
 end
 
 def curl_output(*args)
-  curl_args = curl_args(args) - ["--fail"]
-  Utils.popen_read_text(*curl_args)
+  curl_args = curl_args(args) - ["--fail", "--silent"]
+  Open3.popen3(*curl_args) do |_, stdout, stderr, status|
```

Comment:
At most a few dozen. Mostly things like setting up pipes, then `fork` and `exec` with some massaging of the pipes depending on whether a block was given or not.